### PR TITLE
feat(users): clean tokens on change password

### PR DESCRIPTION
**Pull request by OlegDO**

### CODE REVIEW

File: microservices/users/src/services/change-password.ts [hunk 1]
```diff
@@ -104,10 +122,40 @@ class ChangePassword {
...
+
+      return;
+    }
+
+    // Clear tokens type - REST. Keep only current
+    if (!this.currentToken) {
+      throw new BaseException({
+        status: 500,
+        message: 'Failed to clear rest user tokens. Current user token was not found.',
+      });
```
Comment (MatthewPattell) [inline microservices/users/src/services/change-password.ts]: actualy do we need throw exception? You need to clean up token, but if token not exist, we just skip it

File: microservices/users/src/repositories/user.ts [hunk 2]
```diff
@@ -99,6 +100,63 @@ class User extends Repository<UserEntity> {
...
+        userId,
+        ...(tokenId ? { id: tokenId } : {}),
+      },
+    };
+
+    const { error: countError, result: countResult } = await Api.get().authentication.token.count({
+      query,
+    });
+
+    if (countError || !countResult) {
```
Comment (MatthewPattell) [inline microservices/users/src/repositories/user.ts]: !countResult - just skip operation? Why we need throw error?

File: microservices/users/src/repositories/user.ts [hunk 3]
```diff
@@ -99,6 +100,63 @@ class User extends Repository<UserEntity> {
...
+
+    if (!error) {
+      return;
+    }
+
+    Log.error(error.message);
+
+    throw new BaseException({
+      status: 500,
+      message: 'Failed to remove user tokens.',
```
Comment (MatthewPattell) [inline microservices/users/src/repositories/user.ts:161]: add payload message?

File: microservices/users/src/repositories/user.ts [hunk 4]
```diff
@@ -99,6 +100,63 @@ class User extends Repository<UserEntity> {
...
+    const { error: countError, result: countResult } = await Api.get().authentication.token.count({
+      query,
+    });
+
+    if (countError || !countResult) {
+      Log.error(countError?.message);
+
+      throw new BaseException({
+        status: 500,
+        message: 'Failed to clear rest user tokens.',
```
Comment (MatthewPattell) [inline microservices/users/src/repositories/user.ts:128]: payload message?

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/users/src/repositories/user.ts [hunk 5]
```diff
@@ -99,6 +100,71 @@ class User extends Repository<UserEntity> {
...
+
+    if (countError) {
+      Log.error(countError.message);
+
+      throw new BaseException({
+        status: 500,
+        message: 'Failed to clear rest user tokens.',
+        payload: {
+          userId,
+          currentToken: tokenId,
```
Comment (MatthewPattell) [inline microservices/users/src/repositories/user.ts]: I'm told you about message! countError.message

File: microservices/users/src/repositories/user.ts [hunk 6]
```diff
@@ -99,6 +100,71 @@ class User extends Repository<UserEntity> {
...
+    }
+
+    Log.error(error.message);
+
+    throw new BaseException({
+      status: 500,
+      message: 'Failed to remove user tokens.',
+      payload: {
+        userId,
+        currentToken: tokenId,
```
Comment (MatthewPattell) [inline microservices/users/src/repositories/user.ts]: the same

Comment (MatthewPattell) [reply microservices/users/src/repositories/user.ts]: userId,
        currentToken: tokenId,
        
        is redundant payload

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/users/src/repositories/user.ts (hunk 5 above)
Comment (OlegDO) [reply microservices/users/src/repositories/user.ts]: I got it

**Review by MatthewPattell — APPROVED** [review]

**Comment by MatthewPattell** [comment]
fix sonarcloud before merge

**Pull request merged**